WordPress/gutenberg
# Move HeadinglevelDropdown to its own component

## What?


Updated May 24.

The HeadingLevelDropdown is a toolbar option used to choose between H1-H6 in the heading block, comments title, query title, and post title. The files are in the heading block, and the other blocks import them.
The site title has its own similar option, and it adds a paragraph to the selection.

This PR moves the HeadingLevelDropdown from the heading block to its own component.
Fixes https://github.com/WordPress/gutenberg/issues/45743

## Why?

The same code is reused in multiple blocks.

## How?

Creates a new component that can be consistently reused in multiple blocks.

## Testing Instructions

1. Create a new post and add a post title block.
2. Confirm that the block toolbar has a button with the aria-label "Change level". The visible symbol is H2 which is the default HTML tag for the post title. The previous text for this block was "Change heading level".
3. Select and activate the button (click, press enter or press spacebar). Confirm that this opens a drop down with a button each for the available HTML tags, H1 to H6. There should be an icon and a tooltip with the heading level. 
4. Confirm that the currently selected HTML tag is announced as "checked" (has aria-checked="true") and has a black background.
5. Select and activate the button for one of the heading levels. Confirm that the HTML element for the post title is updated.
6. Save the post and view the front, confirm that the heading level is correct on the front.

## Screenshots or screencast 
Heading block, before (The H1 is hovered over):
<img width="458" alt="Heading block with the level selection opened from the block toolbar" src="https://github.com/WordPress/gutenberg/assets/7422055/ec82486e-1371-4762-a32a-87181ac39d1a">

After, using the updated icons (The H6 is hovered over):
<img width="458" alt="Heading block, with the PR applied" src="https://github.com/WordPress/gutenberg/assets/7422055/d84cd5da-c465-4f39-b07d-d99ad94098e5">
